src/applications/disability-benefits/2346/containers/OrderTable.jsx

```diff
@@ -0,0 +1,186 @@
+import ErrorableCheckbox from '@department-of-veterans-affairs/formation-react/ErrorableCheckbox';
+import React from 'react';
+import { connect } from 'react-redux';
+import { focusElement } from '../../../../platform/utilities/ui';
+import { getDLCData, updateDLCData } from '../actions';
+
+class OrderHistory extends React.Component {
+  // NOTE: Using React state for checkboxes -@maharielrosario at 1/8/2020, 7:47:21 PM
+  // When discussed with other devs, the concensus was to have a stateful
+  // component to house our checkboxes' state. But all other state is in Redux.
+  // Is there an easy way to house individual checkbox state in Redux?
+
+  constructor(props) {
+    super(props);
+    this.createDlcUIData = this.createDlcUIData.bind(this);
+  }
+
+  componentDidMount() {
+    focusElement('.va-nav-breadcrumbs-list');
+    this.props.getDLCData();
+  }
+
+  createDlcUIData(initDataObjs) {
+    let UIDataObjs = initDataObjs;
+
+    (function removeNonUIProps() {
+      for (const UIDataObj of UIDataObjs) {
+        Object.keys(UIDataObj).filter(() => {
+          delete UIDataObj.productGroup;
+          delete UIDataObj.recipient;
+          delete UIDataObj.age;
+          delete UIDataObj.delToState;
+          return UIDataObj;
+        });
+      }
+    })();
+
+    (function createCheckboxProp() {
+      UIDataObjs = UIDataObjs.map(UIDataObj => {
+        const checkbox = (
+          <ErrorableCheckbox
+            className="vads-u-margin-left--2p5"
+            onValueChange={value => value}
+            id={UIDataObj.id}
+          />
+        );
+
+        return {
+          ...UIDataObj,
+          checkbox,
+        };
+      });
+    })();
+
+    const UITableHeaders = Object.keys(UIDataObjs[0]).map(UIDataObjKey => {
```

Comment:
I think instead of having to pair down your objects and then run them through a map/switch case and inject them into a node - you could just destructure out the pieces you need in the render and then add all of the markup to the return - this is the pattern used throughout the rest of the app.  Its also substantially easier to read / understand.

Reply:
https://github.com/department-of-veterans-affairs/vets-website/pull/11532#discussion_r369146197